src/elastio-snap.c

```diff
@@ -5374,6 +5461,8 @@ static asmlinkage long mount_hook(char __user *dev_name, char __user *dir_name,
 		if(!sys_ret) handle_bdev_mounted_writable(dir_name, &idx);
 	}
 
+	if (!sys_ret) bdev_switch_ownership(dir_name, real_flags, OWNERSHIP_TO_DRIVER);
```

Comment:
Should we care about the return value here?

Reply:
About `sys_ret` yes: the idea is to switch owners only if mount/umount was successful. Here we check that mount has worked out properly.